react/features/analytics/middleware.js

```diff
@@ -12,6 +13,16 @@ import { initAnalytics } from './functions';
  */
 MiddlewareRegistry.register(store => next => action => {
     switch (action.type) {
+    case SET_CONFIG: {
+        if (navigator.product === 'ReactNative') {
+            // Reseting the analytics is currently not needed for web because
+            // the user will be redirected to another page and new instance of
+            // Analytics will be created and initialized.
+            resetAnalytics();
+        }
+
+        break;
```

Comment:
Why not do the resetting on web and remove a conditional?

Reply:
Indeed.

Reply:
There are some differences between web and mobile. 

Web:
The set config action is called after the page is loaded. Theoretically there may be cached events that will be send when we initialize the analytics. If we reset the on set config the events will be lost.

Mobile:
The set config action will be called before we navigate the app.